rllab/envs/gym_env.py

```diff
@@ -65,7 +65,11 @@ def __init__(self, env_name, record_video=True, video_schedule=None, log_dir=Non
                 log_dir = os.path.join(logger.get_snapshot_dir(), "gym_log")
         Serializable.quick_init(self, locals())
 
-        env = gym.envs.make(env_name)
+        if isinstance(env, str):
```

Comment:
dispatching on argument type is a Bad Idea.

a couple strategies
* pass a class and args/kwargs for the constructor of that class (e.g. env_cls=None, env_args=[], env_kwargs=dict())
* provide a special constructor function (e.g. GymEnv.fromEnv), probably better

Reply:
I stand corrected. I much closer at this, and this is pretty much the only way to do it while keeping backwards compatibility and not copying a ton of code.